Mathlib/CategoryTheory/Shift/Pullback.lean

```diff
@@ -103,4 +118,47 @@ lemma pullbackShiftFunctorAdd'_hom_app :
     ← Functor.map_comp, Iso.hom_inv_id_app, Functor.map_id]
   rfl
 
+namespace Functor
+
+variable {D : Type*} [Category D] [HasShift D B] (F : C ⥤ D) [F.CommShift B]
+
+/-- If `F : C ⥤ D` commutes with the shifts on `C` and `D`, then it also commutes with
+their pullbacks by an additive map.
+-/
+@[simps]
+noncomputable def commShiftPullback :
+    F.CommShift A (C := PullbackShift C φ) (D := PullbackShift D φ) where
+  iso a := isoWhiskerRight (pullbackShiftIso C φ a (φ a) rfl) F ≪≫
+    CommShift.iso (F := F) (φ a) ≪≫ isoWhiskerLeft _  (pullbackShiftIso D φ a (φ a) rfl).symm
+  zero := by
+    ext
+    simp only [comp_obj, Iso.trans_hom, isoWhiskerRight_hom, isoWhiskerLeft_hom, Iso.symm_hom,
+      NatTrans.comp_app, whiskerRight_app, whiskerLeft_app, CommShift.isoZero_hom_app,
+      pullbackShiftFunctorZero'_hom_app, id_obj, map_comp, pullbackShiftFunctorZero'_inv_app, assoc]
+    conv_lhs => congr; rfl; congr; change (F.commShiftIso (φ 0)).hom.app _
+                rw [F.commShiftIso_zero' (A := B) (φ 0) (by rw [map_zero])]
+    simp only [CommShift.isoZero'_hom_app, assoc]
+  add a b := by
+    ext
+    simp only [comp_obj, Iso.trans_hom, isoWhiskerRight_hom, isoWhiskerLeft_hom, Iso.symm_hom,
+      NatTrans.comp_app, whiskerRight_app, whiskerLeft_app, CommShift.isoAdd_hom_app, map_comp,
+      assoc]
+    conv_lhs => congr; rfl; congr; change (F.commShiftIso (φ (a + b))).hom.app _
+                rw [F.commShiftIso_add' (a := φ a) (b := φ b) (by rw [φ.map_add])]
+    rw [← shiftFunctorAdd'_eq_shiftFunctorAdd, ← shiftFunctorAdd'_eq_shiftFunctorAdd]
+    rw [pullbackShiftFunctorAdd'_hom_app φ _ a b (a + b) rfl (φ a) (φ b) (φ (a + b)) rfl rfl rfl]
+    rw [pullbackShiftFunctorAdd'_inv_app φ _ a b (a + b) rfl (φ a) (φ b) (φ (a + b)) rfl rfl rfl]
+    simp only [CommShift.isoAdd'_hom_app, assoc, comp_obj, map_comp, NatTrans.naturality_assoc,
+      Iso.inv_hom_id_app_assoc]
```

Comment:
```suggestion
    dsimp
    simp only [CommShift.isoAdd_hom_app, map_comp, assoc]
    dsimp
    rw [F.commShiftIso_add' (a := φ a) (b := φ b) (by rw [φ.map_add]),
      ← shiftFunctorAdd'_eq_shiftFunctorAdd, ← shiftFunctorAdd'_eq_shiftFunctorAdd,
      pullbackShiftFunctorAdd'_hom_app φ _ a b (a + b) rfl (φ a) (φ b) (φ (a + b)) rfl rfl rfl,
      pullbackShiftFunctorAdd'_inv_app φ _ a b (a + b) rfl (φ a) (φ b) (φ (a + b)) rfl rfl rfl]
    dsimp
    simp only [CommShift.isoAdd'_hom_app, assoc, map_comp, NatTrans.naturality_assoc,
      Iso.inv_hom_id_app_assoc]
```

Reply:
This suggestion doesn't compile. I did the `conv` + `congr` + `change` line because Lean was refusing to `rw` or even `erw` with `F.commShiftIso_add' (a := φ a) (b := φ b) (by rw [φ.map_add])`, and after your change it is refusing again.

Reply:
It does compile on my computer... (It is dependent on the previous suggestion which changes the `iso` field.)

Reply:
Gah, I thought the previous suggestion was only about removing the `@[simps]`...